data/src/dcc.rs

```diff
@@ -66,12 +66,37 @@ impl Send {
         }
     }
 
-    fn decode<'a>(mut args: impl Iterator<Item = &'a str>) -> Option<Self> {
-        let filename = args.next()?.to_string();
-        let host = args.next().and_then(decode_host)?;
-        let port = NonZeroU16::new(args.next()?.parse().ok()?);
-        let size = args.next()?.parse().ok()?;
-        let token = args.next();
+    fn decode<'a>(args: impl Iterator<Item = &'a str>) -> Option<Self> {
+        let mut args: Vec<&str> = args.collect();
+        args.reverse();
```

Comment:
We can make this bounds a double ended iterator I believe. Though it'd impact the logic below so maybe not.

Reply:
Good idea. Added.